students/sangyeon/users/views.py

```diff
@@ -1,3 +1,60 @@
-from django.shortcuts import render
+import json
+import re
 
+from django.views import View
+from django.http import JsonResponse
+
+from .models import User
+"""
+1. 회원가입을 위한 View 를 작성해야합니다. 사용자 정보는 이름, 이메일, 비밀번호, 연락처(휴대폰), 그 외 개인정보를 포함한다.
+
+2. 이메일이나 패스워드가 전달되지 않을 경우, {"message": "KEY_ERROR"}, status code 400 을 반환합니다.
+
+3. 이메일에는 @와 .이 필수로 포함되어야 합니다. 해당 조건이 만족되지 않은 경우 적절한 에러를 반환해주세요. 이 과정을 Email Validation이라고 합니다. 정규표현식을 활용해주세요. 
+
+4. 비밀번호는 8자리 이상, 문자, 숫자, 특수문자의 복합이어야 합니다. 해당 조건이 만족되지 않은 경우, 적절한 에러를 반환해주세요. 이 과정을 Password Validation이라고 합니다. 정규표현식을 활용해주세요.
+
+5. Email validation, Password Validation 과정에서 정규식을 사용해보세요.
+
+6. 회원가입시 서로 다른 사람이 같은 이메일을 사용하지 않으므로 기존에 존재하는 자료와 중복되면 안됩니다. 적절한 에러를 반환해주세요.
+
+7.회원가입이 성공하면 {"message": "SUCCESS"}, status code 201을 반환합니다.
+"""
 # Create your views here.
+
+class SignupView(View):
+    def post(self, request):
+        input_data = json.loads(request.body)
+
+        name         = input_data["name"]
+        email        = input_data["email"]
+        password     = input_data["password"]
+        phone_number = input_data["phone_number"]
+
+        EGEX_EMAILR = '[a-zA-Z0-9_-]+@[a-z]+.[a-z]+$'
+        REGEX_PASSWORD = '^.*(?=.{8,})(?=.*\d)(?=.*[a-z])(?=.*[@#$%^&+=]).*$'
+        try:
+            if email == "" or password== "":
+                return JsonResponse({"message": "KEY_ERROR"}, status=400)
+
```

Comment:
상연님~ 말씀해주신 것 처럼 KeyError는 다음과 같이 요청으로 함께 들어와야 하는 키 값이 누락되었을 때 나타납니다!

다음은 cilent의 요청 시 KeyError가 아닌 경우와 KeyError인 경우의 예시입니다.
```bash
http -v POST 127.0.0.1:8000/users/signup name="..." email="" password="" phone_number="..."   (KeyError X)
http -v POST 127.0.0.1:8000/users/signup name="..." phone_number="..."   (KeyError O)
```
line 57의 except로 KeyError를 캐치하기 위해서는 try의 범위가 line 29~32를 포함하고 있어야겠죠?

추가로 line 34, 35는 필요없는 코드로 보입니다! 위에서 설명한 것과 같이 `email='' password=''` 같은 경우는 키가 누락되는 KeyError가 아닌 빈값이 들어오는 경우이며 공백이 들어오는 예외 상황은 밑줄의 정규표현식을 통한 유효성 검사 로직에서 걸러지기 때문입니다.

 ```suggestion
    def post(self, request):
        try:
            input_data = json.loads(request.body)

            name         = input_data["name"]
            email        = input_data["email"]
            password     = input_data["password"]
            phone_number = input_data["phone_number"]

            EGEX_EMAILR = '[a-zA-Z0-9_-]+@[a-z]+.[a-z]+$'
            REGEX_PASSWORD = '^.*(?=.{8,})(?=.*\d)(?=.*[a-z])(?=.*[@#$%^&+=]).*$'
 
```

Reply:
수정완료했습니다!